src/coreComponents/fieldSpecification/TractionBoundaryCondition.cpp

```diff
@@ -0,0 +1,221 @@
+
+TractionBoundaryCondition::TractionBoundaryCondition( string const & name, Group * parent ):
+  FieldSpecificationBase( name, parent ),
+  m_tractionType( 0 ),
+  m_inputStress{}//,
+//  m_stressFunctionNames(),
+//  m_useStressFunctions(false),
+//  m_stressFunctions{nullptr}
+{
+  registerWrapper( viewKeyStruct::tractionTypeString(), &m_tractionType ).
+    setApplyDefaultValue( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setSizedFromParent( 0 ).
+    setDescription( "Type of traction boundary condition. Options are:\n"
+                    "0 - Traction is applied to the faces as specified from the scale and direction,\n"
+                    "1 - Traction is applied to the faces as a pressure specified from the product of scale and the outward face normal, \n"
+                    "2 - Traction is applied to the faces as specified by the inner product of input stress and face normal." );
+
+  registerWrapper( viewKeyStruct::inputStressString(), &m_inputStress ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Input stress for tractionType option 2.\n" );
+
+//  registerWrapper( viewKeyStruct::stressFunctionString(), &m_stressFunctionNames ).
+//    setInputFlag( InputFlags::OPTIONAL ).
+//    setDescription( "Function names for description of stress for tractionType "
+//                    "option 2. Overrides m_inputStress.\n" );
+
+
+  getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).
+    setInputFlag( InputFlags::FALSE );
+  setFieldName( catalogName());
+
+  getWrapper< int >( FieldSpecificationBase::viewKeyStruct::componentString() ).
+    setInputFlag( InputFlags::FALSE );
+
+
+}
+
+
+void TractionBoundaryCondition::postProcessInput()
+{
+
+  if( m_tractionType == 0 )
+  {
+    GEOSX_ERROR_IF( LvArray::tensorOps::l2Norm< 3 >( getDirection() ) < 1e-20,
+                    "m_direction is required for m_tractionType==0, but appears to be unspecified" );
+  }
+  else
+  {
+    GEOSX_LOG_RANK_0_IF( LvArray::tensorOps::l2Norm< 3 >( getDirection() ) > 1e-20,
+                         "m_direction is not required unless m_tractionType==0, but appears to be specified" );
+  }
+
+  bool const inputStressRead = getWrapper< R2SymTensor >( viewKeyStruct::inputStressString() ).getSuccessfulReadFromInput();
+  GEOSX_LOG_RANK_0_IF( inputStressRead && m_tractionType!=2,
+                       "m_inputStress is specified, but m_tractionType!=2 so value of m_inputStress is unused." );
+
+  GEOSX_ERROR_IF( !inputStressRead && m_tractionType==2,
+                  "m_tractionType==2, but m_inputStress is not specified." );
+
+
+//  localIndex const numStressFunctionsNames = m_stressFunctionNames.size();
+//  GEOSX_ERROR_IF( numStressFunctionsNames > 0 && numStressFunctionsNames<6,
+//                  "Either 0 or 6 stress functions must be specified using stressFunctions" );
+//
+//  if( numStressFunctionsNames==6 )
+//  {
+//    m_useStressFunctions = true;
+//  }
```

Comment:
I am actually leaving this here intentionally. We have to enable reading the stress input from a space table....which I started to do here, but it would be better for a follow on PR.